### PR TITLE
Asset lineage updated with OpenLineage Services 

### PR DESCRIPTION
**Pull request by DimitriosMaimaris**

### CODE REVIEW

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/GraphVertexMapper.java [hunk 1]
```diff
@@ -27,7 +27,9 @@ public void mapEntityToVertex(LineageEntity lineageEntity, Vertex vertex){
         if (instanceProperties != null) {
 
             for(Map.Entry<String,String> entry: instanceProperties.entrySet()){
-                String key = "ve"+entry.getKey();
+                String key = "veprop"+entry.getKey();
```
Comment (GuyAckermans) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/GraphVertexMapper.java]: Hardcoded String

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/MainGraphMapper.java [hunk 2]
```diff
@@ -4,72 +4,302 @@
...
+     * @param process         - Process to add relationships
+     *
+     * */
+    private void addProcess(Vertex columnInVertex,Vertex columnOutVertex,Vertex process){
+
+        GraphTraversalSource mainG = mainGraph.traversal();
 
         final String processGuid = process.value(PROPERTY_KEY_ENTITY_GUID);
-        final String processName = process.value(PROPERTY_NAME_NAME);
+        final String processName = process.value("vepropdisplayName");
```
Comment (GuyAckermans) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/MainGraphMapper.java]: Hardcoded string, "vepropdisplayName" can be found as constant PROPERTY_KEY_ALTERNATIVE_DISPLAY_NAME in GraphConstants. Maybe it should be "vepropDisplayName"?

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/MainGraphMapper.java [hunk 3]
```diff
@@ -4,72 +4,302 @@
...
+                mainProcess.property(PROPERTY_KEY_DISPLAY_NAME, processName);
+                subProcess.addEdge(EDGE_LABEL_SUBPROCESS_TO_PROCESS,mainProcess);
+
+                mainG.tx().commit();
 
-                vertex.addEdge(NODE_LABEL_PROCESS,mainProcess);
+                addTableNode(columnInVertex,columnOutVertex,mainProcess);
             }
+        }else{
+            log.debug("Process failed");
```
Comment (GuyAckermans) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/MainGraphMapper.java]: Maybe should be log.error, and could use further information saying exactly which process failed

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java [hunk 4]
```diff
@@ -3,34 +3,32 @@
...
+@DisallowConcurrentExecution
 public class BufferGraphJob implements Job {
 
     private static final Logger log = LoggerFactory.getLogger(BufferGraphJob.class);
 
     @Override
     public void execute(JobExecutionContext context) {
         LocalDateTime localTime = LocalDateTime.now();
-        log.info("Run QuartzJob at " + localTime);
+        System.out.println("Run QuartzJob at " + localTime);
```
Comment (GuyAckermans) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java]: Why is System.out preferred over log?

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java [hunk 5]
```diff
@@ -3,34 +3,32 @@
...
     }
 
-    private void performTask(BufferGraph bufferGraph) {
-        try {
-            bufferGraph.schedulerTask();  //TODO  Throws NPE
-        } catch (Exception e) {
-            log.error("Buffergraph job Scheduler error");
-        }
+    private void performTask(BufferGraph bufferGraph){
+        bufferGraph.schedulerTask();
```
Comment (GuyAckermans) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java]: This task can throw an NPE if there is a problem with the buffer graph connection, should be catched.

File: open-metadata-implementation/server-chassis/server-chassis-spring/pom.xml [hunk 6]
```diff
@@ -292,6 +292,11 @@
             <artifactId>slf4j-api</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>janus-connector</artifactId>
+        </dependency>
+
```
Comment (GuyAckermans) [inline open-metadata-implementation/server-chassis/server-chassis-spring/pom.xml]: Should be removed

File: open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/pom.xml [hunk 7]
```diff
@@ -75,6 +75,12 @@
             <scope>compile</scope>
         </dependency>
 
+        <dependency>
```
Comment (planetf1) [inline open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/pom.xml]: The 'server' component MUST NOT have a dependency on spring. We need to be able to support other rest bindings in future (such as microprofile) and keep all spring capability in the spring specific module

File: open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/AssetContextHandler.java [hunk 8]
```diff
@@ -17,8 +17,10 @@
 import org.odpi.openmetadata.repositoryservices.ffdc.exception.RepositoryErrorException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
```
Comment (planetf1) [inline open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/AssetContextHandler.java]: is this only used in the spring layer?

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java [hunk 9]
```diff
@@ -3,34 +3,32 @@
 package org.odpi.openmetadata.governanceservers.openlineage.scheduler;
 
 import org.odpi.openmetadata.governanceservers.openlineage.buffergraph.BufferGraph;
-import org.quartz.Job;
-import org.quartz.JobDataMap;
-import org.quartz.JobExecutionContext;
-import org.quartz.JobKey;
+import org.quartz.*;
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java]: I think it's generally better to keep the specific imports ?

File: open-metadata-implementation/server-chassis/server-chassis-spring/pom.xml [hunk 10]
```diff
@@ -330,6 +330,8 @@
                                 <ignoredUnusedDeclaredDependency>org.springframework.boot:spring-boot-starter-web:*</ignoredUnusedDeclaredDependency>
                                 <ignoredUnusedDeclaredDependency>org.springframework.boot:spring-boot-starter-security:*</ignoredUnusedDeclaredDependency>
                                 <ignoredUnusedDeclaredDependency>org.springframework.boot:spring-boot-starter-data-jpa:*</ignoredUnusedDeclaredDependency>
+                                <ignoredUnusedDeclaredDependency>org.odpi.egeria:janus-connector:*</ignoredUnusedDeclaredDependency>
```
Comment (planetf1) [inline open-metadata-implementation/server-chassis/server-chassis-spring/pom.xml]: I would expect the server layer to depend on janus - not the spring layer?

File: open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/pom.xml (hunk 7 above)
Comment (GuyAckermans) [reply open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/pom.xml]: Agreed

**Pull request merged**